src/service/action-impl.js

```diff
@@ -491,6 +499,51 @@ export class ActionService {
 
 
 /**
+ * A clone of an event object with its function properties noop'd.
+ * This is useful e.g. for event objects that need to be passed to an async
+ * context, but the browser might have cleaned up the original event object.
+ * This clone noops functions since they won't behave normally after the
+ * original object has been destroyed.
+ */
+class DeferredEvent {
+  /**
+   * @param {!Event} event
+   */
+  constructor(event) {
+    cloneWithoutFunctions(event, this);
+
+    /** @type {?Object} */
+    this.detail = null;
```

Comment:
An additional details property was being added to the event object by `addChangeDetails_` and the compiler states you cannot add properties to dict types (which ES6 classes are) after construction. So I added the property in the constructor.

The amount of code mutating native objects in AMP is too damn high! It almost always adds technical debt to do it.